clippy_lints/src/methods.rs

```diff
@@ -1105,6 +1125,50 @@ fn lint_iter_cloned_collect(cx: &LateContext, expr: &hir::Expr, iter_args: &[hir
     }
 }
 
+fn lint_fold_any(cx: &LateContext, expr: &hir::Expr, fold_args: &[hir::Expr]) {
+    // DONOTMERGE: What if this is just some other method called fold?
```

Comment:
Not sure if applicable in this context but other parts of clippy use `match_trait_method(cx, expr, &paths::ITERATOR)`

Reply:
Thanks. I tried this (well, actually, I implemented the equivalent by hand as I missed this method), but this doesn't quite do what I want - it's not quite enough to know that the caller is an Iterator - this could be an inherent impl rather than the trait method. Maybe this is rare/ill-advised enough that it doesn't matter?